### PR TITLE
fix: update outdated links to external resources

### PR DESCRIPTION
**Pull request by krauspt**
## Issue Addressed

I didn't create a specific issue, because changes are trivial, and only documentation is updated.

## Proposed Changes

I've found some outdated links to external resources, this PR fixes them.

## Additional Info

No additional info.

### CODE REVIEW

File: book/src/homebrew.md [hunk 1]
```diff
@@ -31,6 +31,6 @@ Alternatively, you can find the `lighthouse` binary at:
 
 The [formula][] is kept up-to-date by the Homebrew community and a bot that lists for new releases.
 
-The package source can be found in the [homebrew-core](https://github.com/Homebrew/homebrew-core/blob/master/Formula/lighthouse.rb) repo.
+The package source can be found in the [homebrew-core](https://github.com/Homebrew/homebrew-core/blob/master/Formula/l/lighthouse.rb) repo.
```
Comment (chong-he) [inline book/src/homebrew.md:34]: This is fixed in #4822 in the unstable branch but I see no conflict so it should be good

**Review by chong-he — COMMENTED** [review]
Thanks for the fix, looks good to me

File: book/src/homebrew.md (hunk 1 above)
Comment (krauspt) [reply book/src/homebrew.md:34]: Yep, my mistake I should start from the unstable branch instead of stable ...

File: book/src/contributing.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ into the canonical spec.
 ## Rust
 
 Lighthouse adheres to Rust code conventions as outlined in the [**Rust
-Styleguide**](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/guide.md).
+Styleguide**](https://github.com/rust-lang/rust/tree/HEAD/src/doc/style-guide/src).
```
Comment (michaelsproul) [inline book/src/contributing.md]: There's a better rendered version of this style guide here:

```suggestion
Styleguide**](https://doc.rust-lang.org/nightly/style-guide/).
```

**Review by michaelsproul — APPROVED** [review]

**Comment by krauspt** [comment]
@chong-he is there anything from my side that needs to be done?

**Comment by chong-he** [comment]
> @chong-he is there anything from my side that needs to be done?

No, this is ready to be merged once it passes all checks, thanks!

**Review by dapplion — APPROVED** [review]

**Comment by michaelsproul** [comment]
@mergifyio queue

**Pull request merged**